kura/org.eclipse.kura.asset.provider/src/main/java/org/eclipse/kura/asset/provider/BaseAsset.java

```diff
@@ -122,7 +114,7 @@
      * Instantiates a new asset instance.
      */
     public BaseAsset() {
-        this.assetListeners = CollectionUtil.newConcurrentHashMap();
+        this.channelListeners = Collections.synchronizedSet(new HashSet<>());
```

Comment:
I don't think that the registration of channel listeners will be a performance critical task, as you said it will be accessed and updated only occasionally.